metsim/metsim.py

```diff
@@ -247,6 +249,18 @@ def met_data(self):
             self._met_data['elev'] = self.domain['elev']
             self._met_data['lat'] = self.domain['lat']
             self._met_data['lon'] = self.domain['lon']
+
+            # process constant_vars
+            constant_vars = self.params.get('constant_vars', None)
+            if constant_vars:
+                if self.params['forcing_fmt'] != 'binary':
+                    template_var = list(self.params.get('forcing_vars', None).values())[0]
+                else:
+                    template_var = list(self.params.get('forcing_vars', None).keys())[0]
+                for var in constant_vars.keys():
+                    self._met_data[var] = xr.full_like(self._met_data[template_var], 
+                                                       float(constant_vars[var]))
```

Comment:
I would prefer if we did the type-casting in `cli.py` so that it's clear that what should be in `constant_vars[var]` is a float. This makes it stricter for scripting usage, so that users can't specify their constant variables as strings.

Reply:
I would tend to agree, except that is not what is done in the rest of the code either. There are no typecasts in `cli.py` and no meaningful validation of paths and arguments is done before an attempt is made to read the data (resulting in uninformative error messages). 

Currently values are typecast at the time of use (`params['time_step']`) is cast to int repeatedly. So I would like to separate that, i.e. move the `validate` function up (see issue #169 for example) and fail early. 


``` python
 c = int(self.params['chunks'].get(d, s))
 self.disagg = int(self.params['time_step'])
```

Reply:
So I suggest not fixing this now